src/lightning/pytorch/core/datamodule.py

```diff
@@ -158,6 +158,7 @@ def load_from_checkpoint(
         cls,
         checkpoint_path: Union[_PATH, IO],
         hparams_file: Optional[_PATH] = None,
+        map_location=None,  # Add for #17945
```

Comment:
The comment at the back is not desired, we don't track changes like this. 
Could you also add the type information? Thanks!

Reply:
@awaelchli Thanks. It's fixed. Will take care of such practices going onward.